### ISSUE TITLE
Buttons do not contain appropriate accessibility labels

### ISSUE THREAD

**Issue by francisschmaltz** [sourcegraph/sourcegraph#972]
Sourcegraph buttons do not contain any accessibility attributes and this makes it hard to navigate Sourcegraph when using a screen reader. 

Buttons should be descriptive and formatted correctly for screen readers.

All buttons should include the appropriate [ARIA attributes](https://www.w3.org/TR/html-aria/#rules-wd), [role attribute](https://www.w3.org/wiki/PF/XTech/HTML5/RoleAttribute), [title attribute](https://webaim.org/articles/gonewild/), and [ALT attribute](https://accessibility.psu.edu/images/imageshtml/). 

We should make sure that this is inlace by 3.0 as this is a blocker for screen reader usage.

For example: the extensions dropdown button. There is no way for a screen reader to know that this is a button. 
![screen shot 2018-11-13 at 11 34 38 am](https://user-images.githubusercontent.com/11583013/48438175-2af9d100-e738-11e8-8a55-a314951fb994.png)

This button should have at least:
```html
title="Extension dropdown" role="button" aria-label="Extensions dropdown" aria-expanded="false" aria-haspopup="true"
```

**Comment by nicksnyder** [comment]
The first thing that we should do is create a lint rule that identifies all locations where this is a problem.

**Comment by francisschmaltz** [comment]
@felixfbecker can you create a lint rule to catch this.

**Comment by francisschmaltz** [comment]
- [ ] Make sure all buttons are labeled as such @francisschmaltz 
- [ ] Create lint rule to require buttons be labeled @felixfbecker
